Fix xstat prototype to match lstat

This fixes the build with GCC 15.

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>
Upstream-Status: Backport [probably]

diff -durpN cpio-2.15.old/src/extern.h cpio-2.15/src/extern.h
--- cpio-2.15.old/src/extern.h	2024-01-14 12:42:18.000000000 +0100
+++ cpio-2.15/src/extern.h	2025-08-27 09:41:12.357587552 +0200
@@ -97,7 +97,7 @@ extern char input_is_special;
 extern char output_is_special;
 extern char input_is_seekable;
 extern char output_is_seekable;
-extern int (*xstat) ();
+extern int (*xstat) (const char *,  struct stat *);
 extern void (*copy_function) ();
 extern char *change_directory_option;
 
--- cpio-2.15.old/src/global.c	2024-01-14 12:42:18.000000000 +0100
+++ cpio-2.15/src/global.c	2025-08-27 09:47:19.871893208 +0200
@@ -183,10 +183,6 @@
 /* Extract to standard output? */
 bool to_stdout_option = false;
 
-/* A pointer to either lstat or stat, depending on whether
-   dereferencing of symlinks is done for input files.  */
-int (*xstat) ();
-
 /* Which copy operation to perform. (-i, -o, -p) */
 void (*copy_function) () = 0;
 
